oras/provider.py

```diff
@@ -899,9 +901,13 @@ def pull(
 
         files = []
         for layer in manifest.get("layers", []):
-            filename = (layer.get("annotations") or {}).get(
-                oras.defaults.annotation_title
-            )
+            annotations: dict = layer.get("annotations", {})
+            filename = annotations.get(oras.defaults.annotation_title)
+            # A directory will need to be uncompressed and moved
+            unpack_layer = annotations.get(oras.defaults.annotation_unpack, False)
+
+            if unpack_layer and skip_unpack:
```

Comment:
How do we know the filename doesn't already have .tar.gz?

Reply:
After thinking about this for a while: 

Do we even need the `skip_unpack` parameter if we support the official annotation?
For special cases, one can always call `self.download_blob` directly.

To me `unpack_layer and skip_unpack` reads like a contradiction with behavior that is not obvious and must be documented. So I'd rather leave it out. What do you think?